From fbeae3120535212271ddf3de62db61a6726ab2e4 Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <jkysela@redhat.com>
Date: Wed, 3 Sep 2014 07:28:07 -0400
Subject: [alsa] hda: initialize audio InfoFrame to be all zero

Message-id: <140972823573-235-patchset-jkysela@redhat.com>
Patchwork-id: 90935
O-Subject: [RHEL 7.1 PATCH 234/442] ALSA: hda - initialize audio InfoFrame to be all zero
Bugzilla: 1112200
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>

This patch initialized the local audio InfoFrame variable 'ai' to be all zero,
thus the data bytes will indicate "Refer to Stream Header" by default.

Signed-off-by: Mengdong Lin <mengdong.lin@intel.com>
Signed-off-by: Takashi Iwai <tiwai@suse.de>

(picked from kernel: caaf5ef9493f72390905f1e97b310b8906d32dac)
rhbz#1112200
Signed-off-by: Jaroslav Kysela <jkysela@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/sound/pci/hda/patch_hdmi.c b/sound/pci/hda/patch_hdmi.c
index 994713c..4751eab 100644
--- a/sound/pci/hda/patch_hdmi.c
+++ b/sound/pci/hda/patch_hdmi.c
@@ -1064,6 +1064,7 @@ static void hdmi_pin_setup_infoframe(struct hda_codec *codec,
 {
  union audio_infoframe ai;
 
+ memset(&ai, 0, sizeof(ai));
  if (conn_type == 0) { /* HDMI */
   struct hdmi_audio_infoframe *hdmi_ai = &ai.hdmi;
 
-- 
1.7.1